src/edu/washington/escience/myria/tool/MyriaConfiguration.java

```diff
@@ -0,0 +1,261 @@
+package edu.washington.escience.myria.tool;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.commons.io.FilenameUtils;
+import org.ini4j.ConfigParser;
+
+import edu.washington.escience.myria.MyriaConstants;
+import edu.washington.escience.myria.MyriaSystemConfigKeys;
+import edu.washington.escience.myria.accessmethod.ConnectionInfo;
+import edu.washington.escience.myria.coordinator.ConfigFileException;
+
+/** The class to read Myria configuration file, e.g. deployment.cfg. */
+public final class MyriaConfiguration extends ConfigParser {
+
+  /** */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * load the config file.
+   * 
+   * @param filename filename.
+   * @return parsed mapping from sections to keys to values.
+   * @throws ConfigFileException if error occurred when parsing the config file
+   * */
+  public static MyriaConfiguration loadWithDefaultValues(final String filename) throws ConfigFileException {
+    File f = new File(filename);
+    if (!f.exists()) {
```

Comment:
Why not just let a `FileNotFoundException` be thrown later when you try to read the file?